dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -47,63 +62,70 @@ public void execute(final Namespace ns, final DockerfileGitHubUtil dockerfileGit
             // the org gets included in the search query.
             orgsToIncludeInSearch.put(ns.get(Constants.GIT_ORG), true);
         }
+        Optional<Exception> failureMessage;
         List<ProcessingErrors> imagesThatCouldNotBeProcessed = new LinkedList<>();
-        AtomicInteger numberOfImagesToProcess = new AtomicInteger();
-        for (Map.Entry<String, JsonElement> imageToTag : imageToTagStore) {
-            numberOfImagesToProcess.getAndIncrement();
-            String image = imageToTag.getKey();
-            String tag = imageToTag.getValue().getAsString();
-            try {
-                PullRequests pullRequests = getPullRequests();
-                GitHubPullRequestSender pullRequestSender = getPullRequestSender(dockerfileGitHubUtil, ns);
-                GitForkBranch gitForkBranch = getGitForkBranch(image, tag, ns);
-
-                log.info("Finding Dockerfiles with the image name {}...", image);
-
-                Optional<List<PagedSearchIterable<GHContent>>> contentsWithImage =
-                        this.dockerfileGitHubUtil.findFilesWithImage(image, orgsToIncludeInSearch, gitApiSearchLimit);
-
-                if (contentsWithImage.isPresent()) {
-                    contentsWithImage.get().forEach(pagedSearchIterable -> {
-                        try {
-                            pullRequests.prepareToCreate(ns, pullRequestSender,
-                                    pagedSearchIterable, gitForkBranch, dockerfileGitHubUtil);
-                        } catch (IOException e) {
-                            log.error("Could not send pull request for image {}.", image);
-                            processErrors(image, tag, e, imagesThatCouldNotBeProcessed);
-                        }
-                    });
+        for (ImageTagStoreContent content : imageNamesWithTag) {
+            String image = content.getImageName();
+            String tag = content.getTag();
+            failureMessage = processImageWithTag(image, tag, ns, orgsToIncludeInSearch, gitApiSearchLimit);
+            if (failureMessage.isPresent()) {
+                ProcessingErrors processingErrors = processErrorMessages(image, tag, failureMessage);
+                imagesThatCouldNotBeProcessed.add(processingErrors);
+            }
```

Comment:
This is a good place for `ifPresent`, https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-

```
failureMessage.ifPresent(message -> imagesThatCouldNotBeProcessed.add(processErrorMessages(image, tag, message)));
```